pkg/flowaggregator/flowaggregator.go

```diff
@@ -15,25 +15,257 @@
 package flowaggregator
 
 import (
+	"fmt"
+	"hash/fnv"
 	"net"
 	"time"
 
-	"github.com/vmware/go-ipfix/pkg/registry"
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+	ipfixintermediate "github.com/vmware/go-ipfix/pkg/intermediate"
+	ipfixregistry "github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/ipfix"
+	"github.com/vmware-tanzu/antrea/pkg/util/env"
+)
+
+var (
+	ianaInfoElements = []string{
+		"flowStartSeconds",
+		"flowEndSeconds",
+		"sourceTransportPort",
+		"destinationTransportPort",
+		"protocolIdentifier",
+		"packetTotalCount",
+		"octetTotalCount",
+		"packetDeltaCount",
+		"octetDeltaCount",
+		"sourceIPv4Address",
+		"destinationIPv4Address",
+	}
+	ianaReverseInfoElements = []string{
+		"reversePacketTotalCount",
+		"reverseOctetTotalCount",
+		"reversePacketDeltaCount",
+		"reverseOctetDeltaCount",
+	}
+	antreaInfoElements = []string{
+		"sourcePodName",
+		"sourcePodNamespace",
+		"sourceNodeName",
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+		"destinationServicePort",
+		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
+		"egressNetworkPolicyName",
+		"egressNetworkPolicyNamespace",
+		"destinationClusterIPv4",
+	}
+	aggregatorElements = []string{
+		"originalExporterIPv4Address",
+		"originalObservationDomainId",
+	}
+	aggregationFields = []string{
+		"destinationPodName",
+		"destinationPodNamespace",
+		"destinationNodeName",
+		"destinationClusterIPv4",
+		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
+	}
+	aggregationWorkerNum = 2
+)
+
+type AggregatorTransportProtocol string
+
+const (
+	AggregatorTransportProtocolTCP AggregatorTransportProtocol = "TCP"
+	AggregatorTransportProtocolUDP AggregatorTransportProtocol = "UDP"
 )
 
 type flowAggregator struct {
-	flowCollectorAddr net.Addr
-	exportInterval    time.Duration
+	externalFlowCollectorAddr   net.Addr
+	aggregatorTransportProtocol AggregatorTransportProtocol
+	collectingProcess           ipfix.IPFIXCollectingProcess
+	aggregationProcess          ipfix.IPFIXAggregationProcess
+	exportInterval              time.Duration
+	exportingProcess            ipfix.IPFIXExportingProcess
+	templateID                  uint16
+	registry                    ipfix.IPFIXRegistry
 }
 
-func InitFlowAggregator(flowCollectorAddr net.Addr, exportInterval time.Duration) *flowAggregator {
+func NewFlowAggregator(externalFlowCollectorAddr net.Addr, exportInterval time.Duration, aggregatorTransportProtocol AggregatorTransportProtocol) *flowAggregator {
+	registry := ipfix.NewIPFIXRegistry()
 	registry.LoadRegistry()
-	return &flowAggregator{
-		flowCollectorAddr,
+	fa := &flowAggregator{
+		externalFlowCollectorAddr,
+		aggregatorTransportProtocol,
+		nil,
+		nil,
 		exportInterval,
+		nil,
+		0,
+		registry,
+	}
+	return fa
+}
+
+func genObservationID() (uint32, error) {
+	name := env.GetPodName()
```

Comment:
I think my request to add a comment explaining why this is a good choice to generate the observationID is still valid. In particular, with this value, the observationID will change if the Pod is killed (for any reason) and re-scheduled.

Reply:
Yes, you are right. Observation Domain ID should be unique per cluster during its life-time. I think cluster name might be a good choice?

Reply:
I looked around. There is no concept of cluster name or unique cluster ID in k8s--looks like the best we could is UID of kube-system namespace: https://groups.google.com/g/kubernetes-sig-architecture/c/mVGobfD4TpY/m/nkdbkX1iBwAJ?pli=1
This requires adding k8s go-client support in the aggregator. 

Maybe for now we can consider the DNS name or just the static service name "flow-aggregator" and move on?

Reply:
For the telemetry implementation, I am working on a cluster UUID, so you may be able to piggy back on that when it is available. In the mean time, I think you can use whatever works best and does not require too much effort, and add a comment. I believe an observationID that changes when the Pod is restarts is acceptable for now, and is better than an observationID that is shared between multiple clusters connected to the same flow collector?

Reply:
Oh great. Cluster UUID will be good. 
Regarding a dynamic observationID, I guess this depends on what flow collector will be using the observationID for.
@zyiou From the perspective of ELK flow collector, what observationID is good here?

Reply:
Any signed64 integer will work for elk flow collector. reference [here](https://www.elastic.co/guide/en/beats/filebeat/current/exported-fields-netflow.html#_exporter) `netflow.exporter.source_id` part.

Reply:
From an offline chat with @zyiou, we are not using ObservationID for any filtering in the elk-flow-collector. 

@antoninbas As we are currently only supporting elk-flow-collector inside the cluster and not claiming any multiple cluster support, is it better to have the static name for now in case we support filtering in elk-flow-collector? I am looking at multiple cluster support as something that comes much further in the future.

Reply:
Thanks, using the DNS name as the temporary solution, will change to use cluster UUID.